foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_7_0/EnableDistOutputFolderForHabushuBuildCacheMigration.java

```diff
@@ -33,23 +33,31 @@ private boolean isHabushuPackaged(Model model) {
     }
 
     private boolean doesBuildDirTagExist(Model model) {
-        return null != model.getBuild().getLocation("directory");
+        return null != model.getBuild() && null != model.getBuild().getLocation("directory");
     }
 
     @Override
     protected boolean shouldExecuteOnFile(File file) {
         Model model = getLocationAnnotatedModel(file);
-        return !doesBuildDirTagExist(model) && isHabushuPackaged(model);
+        return isHabushuPackaged(model) && !doesBuildDirTagExist(model);
     }
 
     @Override
     protected boolean performMigration(File file) {
         Model model = getLocationAnnotatedModel(file);
         Build build = model.getBuild();
-        InputLocation buildLocation = build.getLocation("");
-        InputLocation inputLocation = new InputLocation(buildLocation.getLineNumber() + 1, buildLocation.getColumnNumber());
-        String line = "\t\t<directory>dist</directory>\n";
         int indent = 2;
+        InputLocation inputLocation;
+        String line;
+        if (build != null) {
+            InputLocation referenceLocation = build.getLocation("");
+            inputLocation = new InputLocation(referenceLocation.getLineNumber() + 1, referenceLocation.getColumnNumber());
+            line = "\t\t<directory>dist</directory>\n";
+        } else {
+            InputLocation referenceLocation = model.getLocation("packaging");
```

Comment:
A: This feels like the wrong location given the test POM and our usual layout in our MDA templates.  Makes more sense to me to do it before `dependencies` at the very end before the `</project>` line.  The former feels a bit more natural, but the latter is less complex as dependencies could be missing.